students/EricAdams/session03/mailroom.py

```diff
@@ -0,0 +1,133 @@
+#!/usr/bin/env python
+
+# Data
+donors_and_donations = ['Steve Jobs', [5, 10, 15], 'Bill Gates', [
+    30, 40, 50], 'Barak Obama', [50], 'Donald Trump', [
+    1, 2, 3], 'Melinda Gates', [10, 20, 30]]
+#
+# “Send a Thank You”, “Create a Report” or “quit”
+#
+
+
+def create_a_report():
+    """ Print out a report of donors, total given number of donations
+     and average donation
+    """
+    donor_name = []
+    donor_total_given = []
+    donor_total_num_of_gifts = []
+    total_given = 0.0
+    average_gift = []
+    list_of_donations = []
+    average_of_gifts = 0
+    print('{:<20}|{:^15}|{:^15}|{:^15}'.format(
+        'Donor Name', 'Total Given', 'Num Gifts', 'Average Gift'))
+    print('-' * 65)
+    # Separate the individual donors and donations into separate lists
+    i = 0
+    while i < len(donors_and_donations):
+        donor_name.append(donors_and_donations[i])
+        list_of_donations.append(donors_and_donations[i + 1])
+        i += 2
+    # Calculate the sum of the gifts, the number of gifts, avg gift amount
+    # for each donor, (each in a separate list)
+    for donations in list_of_donations:
+        # print(donations)
+        # print(len(donations))
+        i = 0
+        total_given = 0
+        # calculate the total given
+        while i < len(donations):
+            # print(donations[i])
+            total_given += donations[i]
+            i += 1
+        average_of_gifts = total_given / len(donations)
+        average_gift.append(average_of_gifts)
+        donor_total_given.append(total_given)
+        donor_total_num_of_gifts.append(len(donations))
+    i = 0
+    while i < len(donors_and_donations) // 2:
+        print('{:<20}${:^15}${:^15}${:8.2f}'.format(
+            donor_name[i], donor_total_given[i], donor_total_num_of_gifts[i],
+            average_gift[i]))
+        i += 1
+
+
+def compose_email(answer, answer1):
```

Comment:
answer, answer1 are odd names for these parameters -- maybe:

``compose_email(name, amount)`` ?